server/pulp/server/async/tasks.py

```diff
@@ -255,7 +255,7 @@ def apply_async_with_reservation(self, resource_type, resource_id, *args, **kwar
         # To avoid the race condition where __call__ method below is called before
         # this change is propagated to all db nodes, using an 'upsert' here and setting
         # the task state to 'waiting' only on an insert.
-        task_status.save(fields_to_set_on_insert=['state', 'start_time'])
+        task_status.save(fields_to_set_on_insert=['state', 'start_time'], safe=True)
```

Comment:
@rbarlow In the case of this specific PR, it introduces `safe=True` so it was not being used before here.

Also the `safe` parameter is not deprecated, it's literally missing from the most current version of mongoengine. That means in other areas of code when we set that option, we would expect it to be committed synchronously but the option will be ignored entirely and won't provide the expected behavior. I'm suggesting `write_concern={fsync: True}` as an alternative because it seems to provide that expected behavior which is "don't return until the write has been committed".

I propose we introduce the setting of the `w` option throughout the entire codebase as part of issue 1, or maybe a separate issue itself. It's such a broad problem that I don't think there is much value in setting `w` for this case when its still such a problem elsewhere.